src/xvectorbin/nnet3-xvector-signal-perturb-egs.cc

```diff
@@ -73,16 +73,28 @@ void ApplyPerturbation(XvectorPerturbOptions opts,
   }
   // Perturb speed of signal egs
   Matrix<BaseFloat> warped_distorted_shifted_egs(rand_distort_shifted_egs);
-  if (opts.max_time_stretch != 0.0) 
-    perturb_xvector.TimeStretch(rand_distort_shifted_egs, 
-                                &warped_distorted_shifted_egs);
+  ////if (opts.max_time_stretch != 0.0) 
+  ////  perturb_xvector.TimeStretch(rand_distort_shifted_egs, 
+  ////                              &warped_distorted_shifted_egs);
    
   // If nagation is true, the sample values are randomly negated
   // with some probability.
-  if (opts.negation) {
+  ////if (opts.negation) {
    
-  }
+  ////}
+}
+
+// add
+// This function is a entrance. It calls ApplyDistortion to apply different
+// type of distortions on input.
+void PerturbExample(XvectorPerturbOptions opts,
+                    const Matrix<BaseFloat> &input_egs,
+                    Matrix<BaseFloat> *perturb_egs) {
+  //new a PerturbXvectorSignal object and call ApplyDistortion
+  PerturbXvectorSignal perturb_xvector(opts);
```

Comment:
1. Change perturb_egs to perturbed_eg
2. Change perturb_xvector  to perturb_egs.